themes/default/content/blog/kubernetes-architecture-templates/index.md

```diff
@@ -114,6 +116,8 @@ description: A Python program to deploy a Kubernetes cluster and application on
 runtime: python
 ```
 
+{{< /code-filename >}}
+
 And now we can set up our new `__main__.py` file! Create a new `__main__.py` file in the root of the repo, and add this code:
 
 {{< code-filename file="<root>/__main__.py" >}}
```

Comment:
I believe the intent of these shortcodes was to have them not wrap content, but just be standalone ("self-closing") and sit above the blocks they refer to. Could you try this instead and see if that works?

```suggestion
{{< code-filename file="<root>/__main__.py" />}}
```

Reply:
So I just tried that. It will build, but there's a difference in output:
the shortcode [uses `.Inner`](https://github.com/pulumi/pulumi-hugo/blob/c5c8c95354257403c9e012ae7317903ccd83dc72/themes/default/layouts/shortcodes/code-filename.html#L6-L8) to refer to the contents between the open and close.
With self-closing, `.Inner` is empty.

The effect of this is that without closing after the code block, the code block is a separate div after the file name,
whereas with closing after the code block, the code block is part of the div that holds the file name.

That's not an issue necessarily, because based on current styles, that has no difference visually, only in the document internals.
But if the style changed to, say, draw a border around the filename-code-block combo, the code block would not be correctly associated with the file name.

Since that's just a hypothetical, I don't have a strong preference either way.
I'll defer to your choice of using self-closing or explicitly closed tags here.

Reply:
Ok cool, thanks for trying that out -- and good to know what the changes to the markup would be. Let's stick with self-closing for now, then, since that's closer to what was originally intended. (I'd also like retire this shortcode at some point soonish in favor of something a little more Markdown-specific.)

Reply:
Sounds good! Updated.